packages/application-shell-connectors/src/components/application-context/application-context.tsx

```diff
@@ -73,9 +73,9 @@ type TRawProject = {
   owner: {
     id: string;
   };
-  permissions: TApplicationContextPermissions;
-  actionRights: TApplicationContextActionRights;
-  dataFences: TApplicationContextDataFences;
+  permissions: TApplicationContextPermissions | null;
+  actionRights: TApplicationContextActionRights | null;
+  dataFences: TApplicationContextDataFences | null;
```

Comment:
All those should be either defined or null.

Reply:
For instance, if we pass an empty object to `dataFences` it's expected that the object has at least a field named `store`. So basically it does not make sense to pass empty objects.

Reply:
"it does not make sense" sorry I did doubt sense. I do not dare to counter argue now 😄